src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJobInd.java

```diff
@@ -0,0 +1,106 @@
+
+import edu.ucsb.cs156.happiercows.entities.Commons;
+import edu.ucsb.cs156.happiercows.entities.Profit;
+import edu.ucsb.cs156.happiercows.entities.User;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import edu.ucsb.cs156.happiercows.repositories.CommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.ProfitRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserCommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserRepository;
+import edu.ucsb.cs156.happiercows.services.jobs.JobContext;
+import edu.ucsb.cs156.happiercows.services.jobs.JobContextConsumer;
+import edu.ucsb.cs156.happiercows.errors.EntityNotFoundException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import java.util.Optional;
+
+@AllArgsConstructor
+public class MilkTheCowsJobInd implements JobContextConsumer {
+
+    @Getter
+    private CommonsRepository commonsRepository;
+    @Getter
+    private UserCommonsRepository userCommonsRepository;
+    @Getter
+    private UserRepository userRepository;
+    @Getter
+    private ProfitRepository profitRepository;
+    @Getter
+    private long commonsID;
+
+    public String formatDollars(double amount) {
+        return  String.format("$%.2f", amount);
+    }
+
+    @Override
+    public void accept(JobContext ctx) throws Exception {
+        ctx.log("Starting to milk the cows");
+        Optional<Commons> commonMilkedOpt = commonsRepository.findById(commonsID);
+
+        if(commonMilkedOpt.isPresent()){
+            Commons commonMilked = commonMilkedOpt.get();
+            String name = commonMilked.getName();
+            double milkPrice = commonMilked.getMilkPrice();
+            ctx.log("Milking cows for Commons: " + name + ", Milk Price: " + formatDollars(milkPrice));
+
+            Iterable<UserCommons> allUserCommons = userCommonsRepository.findByCommonsId(commonMilked.getId());
+
+            for (UserCommons userCommons : allUserCommons) {
+                milkCows(ctx, commonMilked, userCommons);
+            }
+            
+
+            ctx.log("Cows have been milked!");
+        } else {
+            ctx.log(String.format("No commons found for id %d", commonsID));
+        }
+    }
+
+    /** This method performs the function of milking the cows for a single userCommons.
+     *  It is a public method only so it can be exposed to the unit tests
+     * @param ctx the JobContext
+     * @param commons the Commons
+     * @param userCommons the UserCommons
+     *
+     */
+
+    public void milkCows(JobContext ctx, Commons commons, UserCommons userCommons) {
+        User user = userCommons.getUser();
+
+        ctx.log("User: " + user.getFullName()
+                + ", numCows: " + userCommons.getNumOfCows()
+                + ", cowHealth: " + userCommons.getCowHealth()
+                + ", totalWealth: " + formatDollars(userCommons.getTotalWealth()));
+
+        double profitAmount = calculateMilkingProfit(commons, userCommons);
```

Comment:
```suggestion
        double profitAmount = MilkTheCows.calculateMilkingProfit(commons, userCommons);
```

At the very least, don't create a second version of this method when there's already one you can use.